File: lua/ht/plugins/ui/nvim-notify.lua
return {
  -- Better `vim.notify()`
  Use {
    "rcarriga/nvim-notify",
    lazy = {
      dependencies = { "nvim-tree/nvim-web-devicons" },
      lazy = true,
      opts = {
        timeout = 3000,
        stages = "static",
        fps = 60,
        max_height = function()
          return math.floor(vim.o.lines * 0.75)
        end,
        max_width = function()
          return math.floor(vim.o.columns * 0.75)
        end,
      },
      config = true,
    },
    functions = {
      FuncSpec("List notify histories using telescope", function()
        require("telescope").extensions.notify.notify()
      end, {
        keys = "<leader>lh",
        desc = "notify-history",
      }),
      FuncSpec("Dismiss all notifications", function()
        require("notify").dismiss { silent = true, pending = true }
      end, {
        keys = "<leader>nn",
        desc = "notify-history",
      }),
    },
  },
}

File: src/conf/plugins/ui/nvim-notify.ts
import { Plugin, PluginOpts } from "@core/model";

const spec: PluginOpts = {
  shortUrl: "rcarriga/nvim-notify",
  lazy: {
    dependencies: ["nvim-tree/nvim-web-devicons"],
    lazy: true,
    opts: {
      timeout: 3000,
      stages: "static",
      fps: 60,
      max_height: () => {
        return Math.floor(
          vim.api.nvim_get_option_value("lines", {
            scope: "global",
          }) * 0.75
        );
      },
      max_width: () => {
        return Math.floor(
          vim.api.nvim_get_option_value("columns", {
            scope: "global",
          }) * 0.75
        );
      },
    },
    config: (_, opts) => {
      vim.defer_fn(() => {
        luaRequire("notify").setup(opts);
      }, 30);
    },
  },
  extends: {
    commands: [
      {
        name: "List notify histories using telescope",
        callback: () => {
          luaRequire("telescope").extensions.notify.notify();
        },
        keys: "<leader>lh",
        shortDesc: "notify-history",
      },
      {
        name: "Dismiss all notifications",
        callback: () => {
          luaRequire("notify").dismiss({
            silent: true,
            pending: true,
          });
        },
        keys: "<leader>nn",
        shortDesc: "notify-dismiss",
      },
    ],
  },
};

export const plugin = new Plugin(spec);
